### PR TITLE
Add 'Has interviews' filter to support console

### PR DESCRIPTION
**Pull request by mnacos**
## Context

For supporting and tracking usage of interviews feature.

## Changes proposed in this pull request

Add a 'Has interviews' filter to support console.

image

Add support for `interviewing` state to `ProcessState`. This adds the following segment to our process docs:

image

## Guidance to review

Navigate to support, use the filter.

## Link to Trello card

https://trello.com/c/1fkhJzrW

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/models/support_interface/applications_filter.rb [hunk 1]
```diff
@@ -25,6 +25,10 @@ def filter_records(application_forms)
         application_forms = application_forms.where('phase IN (?)', applied_filters[:phase])
       end
 
+      if applied_filters[:interviews]
+        application_forms = application_forms.joins(application_choices: [:interviews])
```
Comment (duncanjbrown) [inline app/models/support_interface/applications_filter.rb:29]: Could we perhaps have a test for this one? Just a unit test for the filter. I added a similar one in #3967

**Review by duncanjbrown — COMMENTED** [review]
LGTM, could have 1 small test!

File: app/models/support_interface/applications_filter.rb (hunk 1 above)
Comment (mnacos) [reply app/models/support_interface/applications_filter.rb:29]: Ok, added

**Review by duncanjbrown — APPROVED** [review]
Thank you!

**Review by AgaDufrat — APPROVED** [review]
The only non-dev question I have is about having a filter with just 1 checkbox. 
I've done it myself #3400 but later on @steventux had a chat with the content/design team and went with 2 checkboxes: https://github.com/DFE-Digital/apply-for-teacher-training/pull/3680#issuecomment-749519784

Nothing blocking but I'd add a card to Tech debt to make this consistent across the app.

**Comment by mnacos** [comment]
> The only non-dev question I have is about having a filter with just 1 checkbox.
> I've done it myself #3400 but later on @steventux had a chat with the content/design team and went with 2 checkboxes: [#3680 (comment)](https://github.com/DFE-Digital/apply-for-teacher-training/pull/3680#issuecomment-749519784)
> 
> Nothing blocking but I'd add a card to Tech debt to make this consistent across the app.

Thanks for pointing this out Aga! I'll merge as is and we could revisit. In my mind, this is a temporary filter for tracking feature usage post-launch and we could decide to remove it after a while.

**Pull request merged**